ffi/diplomat/src/calendar.rs

```diff
@@ -32,4 +37,56 @@ pub mod ffi {
                 .into()
         }
     }
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::calendar::AnyCalendar, Enum)]
+    pub struct ICU4XCalendar(pub Arc<AnyCalendar>);
```

Comment:
Question: why `ICU4XCalendar` and not `ICU4XAnyCalendar`? How do you anticipate us representing typed calendars and dates in FFI?

Reply:
ICU4XJapaneseCalendar etc

"Any" is a Rusty term and only really makes sense to me when there's a generic to compare against, which we don't have here.

Also I'd imagine that we won't actually have individual calendars exposed over FFI and just use AnyCalendar for everything. It's far nicer over FFI and it follows our history of using dynamic dispatch to deal with generics over FFI.

Reply:
(I don't have a strong opinion here though)